draft/mailing_list.rst

```diff
@@ -0,0 +1,19 @@
+2. Mailing lists are now only detected by the ListId header, because how should DC sort mailing lists if the ListId header is unset and only the precedence header says it's a mailing list (?). We could just hide these emails (with "junk" or "list" precedence), as we did before, if there is a reason to do so. 
+
+   I do not actually understand why such emails were hidden in the first place, though; if there is an automatic answer stating that someone is out of office or if an email could not be delivered (these are the usual reasons why such emails are sent), I would want to know about this as a user.
+   
+  Björn: to the ListId-header: it is already an improvement to use only that and to keep ignoring mails with Precedence-header. historically, i ignored all these mails to reduce noise that is created from them (eg. contacts should not be added to the suggestion list...) and to be able to concentrate on other things first.
+   
+   
+3. for `List-Id: foo <bla>`, bla now is the id and foo is the name. For GitHub and GitLab notifications this is bad because all notifications will go into one chat. Maybe we should instead take the "in-reply-to" header to find out what messages belong together. For normal mailing lists, of course, this will create a second chat if someone does not use the "Reply" button to write to that mailing list.
+
+  Björn: well, having all notifications in one chat is not that bad. i would just follow the guidelines for now, use the ID and not the Name.
+
+
+4. Currently a mailing list is shown as an empty group (ChatType `Group`) ("0 members"). 
+
+   Maybe we should change the ChatType to `Single` because this way, the UI would fit better. Disadvantage: We can't show the different senders of the messages. 
+   
+   Or we introduce a `ChatType` `MailingList`. Very big disadvantage: We would have to adapt all UI project and it would be nice if we could keep the changes within the core.
```

Comment:
Why can a mailing list not be a group of two members -- the mailing list address and myself? 
Would this not still allow showing other ML-members as senders (even with their avatar)?

Reply:
Would that not be ["impersonation"](https://support.delta.chat/t/bot-impersonate-alias-names/482), which we didn't yet implement?

Reply:
We should also take into consideration that a lot of mailing lists these days use the lists's email-address as addr-spec in `From` (to circumvent DMARC-issues). We could only look at `Reply-To` in order to find the probable original sender, in these cases.

Reply:
> Would this not still allow showing other ML-members as senders (even with their avatar)?

UX-wise that could look nice at first sight, but I think it might lead to confusion, too: Seeing individual "senders" of messages gives me the idea that I could have a look who's "in the group". But I can't, which is inconsistent behaviour.

I think we should only show the individual senders of a message if we also can list all subscribers as group-members. (The list of members must be read-only then.)

Reply:
> Why can a mailing list not be a group of two members -- the mailing list address and myself?
> Would this not still allow showing other ML-members as senders (even with their avatar)?

Yes, it should be possible. But I agree with @pabzm that it might be confusing to users. 

On the other hand, the other solutions I proposed are not really better, either. However, I would make it a group with only one member, the mailing list address. Seems logical/fair to me that if the other _mailing list members_ are not shown in the list, I should not be shown, either. Also, I personally think that "1 member" is less confusing than "0 members" or "2 members".

> We should also take into consideration that a lot of mailing lists these days use the lists's email-address as addr-spec in From (to circumvent DMARC-issues). We could only look at Reply-To in order to find the probable original sender, in these cases.

How could `Reply-To` help? But you are right that it's a problem, however, I will try to use the name (`holger krekel <notifications@github.com>`, not `notifications@github.com`).

What is difficult is finding out whether `From:` is the mailing list address or the actual sender's address. Because in the latter case we want to use only the actual address as usual so that we can show the correct avatar. I think that I will try comparing the `From:` domain with the `List-Id` domain.

> I think we should only show the individual senders of a message if we also can list all subscribers as group-members. (The list of members must be read-only then.)

We can't.

Reply:
> How could `Reply-To` help?

If the list's email address is used as `From` addr-spec, then usually the original `From` is added as `Reply-To`. That's not a safe bet, but it is valid most of the time.

Reply:
> > How could `Reply-To` help?
> 
> If the list's email address is used as `From` addr-spec, then usually the original `From` is added as `Reply-To`. That's not a safe bet, but it is valid most of the time.

Do you have such a mailing list email? Can you send it to me? 

Or we just concentrate on using the whole address (holger krekel <notifications@github.com>, not notifications@github.com). This will have the advantage of working with GitHub/GitLab notifications.

Reply:
> Do you have such a mailing list email?

https://begriffs.com/posts/2018-09-18-dmarc-mailing-list.html has an example of "From-munging" (search for "Here is what From-munging looks like").